components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -1703,8 +1703,7 @@ private String constructAssertionConsumerUrl(MessageContext messageContext) {
         //Note: Do not change to construct the assertionConsumerUrl directly using servicePrefix instead of headers.get("HOST").
         //It always gives IP for host which cause invalid assertionConsumerUrl
         String assertionConsumerUrl = servicePrefix.substring(0, servicePrefix.indexOf("/") + 2) +
-                                      headers.get("HOST") + messageContext.getProperty("REST_API_CONTEXT") +
-                                      "/" + messageContext.getProperty("SYNAPSE_REST_API_VERSION") + "/";
+                headers.get("HOST") + messageContext.getProperty("REST_API_CONTEXT") + "/";
 
```

Comment:
Cannot use pre-defined constant `AppMConstants.HOST` here, because its value is "[Host](https://github.com/wso2/carbon-appmgt/blob/63a8f872440f1bd77e34d1cb630db903dd075299/components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/AppMConstants.java#L590)", not "HOST".
fixed in commit 938f3cb491db8c72b33cb1c5e1b438311bc44acf of PR #153